pyethapp/eth_service.py

```diff
@@ -169,11 +158,17 @@ def __init__(self, app):
         coinbase = app.services.accounts.coinbase
         env = Env(self.db, sce['block'])
 
-        genesis_data = sce['genesis_data'] if 'genesis_data' in sce else dict()
-        self.chain = Chain(env=env,
-                           genesis=genesis_data,
-                           coinbase=coinbase,
-                           new_head_cb=self._on_new_head)
+        genesis_data = sce.get('genesis_data', {})
+        if not genesis_data:
+            genesis_data = mk_genesis_data(env)
+            # XXX: Workaround for
+            # https://github.com/ethereum/pyethereum/issues/726
+            genesis_data["extraData"] = genesis_data["extra_data"]
+            genesis_data["parentHash"] = genesis_data["prevhash"]
+            genesis_data["gasLimit"] = genesis_data["gas_limit"]
```

Comment:
Do you think we can fix https://github.com/ethereum/pyethereum/blob/develop/ethereum/genesis_helpers.py#L81 to generate camelCase keys and remove L164-L168?